ocamldoc/odoc_ast.ml

```diff
@@ -725,12 +725,12 @@ module Analyser =
                     Odoc_messages.object_end
           in
           let param_exps = List.fold_left
-              (fun acc -> fun (_, exp_opt) ->
-                match exp_opt with
-                  None -> acc
-                | Some e -> acc @ [e])
+              (fun acc -> fun (_, arg) ->
+                match arg with
+                | Omitted () -> acc
+                | Arg e -> acc @ [e])
               []
-              exp_opt_optional_list
+              arg_list
           in
```

Comment:
isn't that just `List.filter (fun (_, arg) -> not (is_omitted arg)) arg_list` (provided you carry the definition of `is_omitted` here?

Reply:
Actually, `filter_map` is the fitting one here.